x/dclauth/types/message_propose_add_account.go

```diff
@@ -61,6 +65,21 @@ func (msg *MsgProposeAddAccount) HasRole(targetRole AccountRole) bool {
 	return false
 }
 
+func (msg *MsgProposeAddAccount) HasValidProductIDs() error {
+	var lastMax int32
+	for _, productRange := range msg.ProductIDs {
+		if productRange.Min > productRange.Max {
+			return fmt.Errorf("Uint16Range.Max should be greater or equal to Uint16Range.Max: min=%d, max=%d", productRange.Min, productRange.Max)
```

Comment:
You current message suggests that the Uint16Range.Max value is compared to itself, which is likely typo.
I think a more correct message would be  'PID max value must be greater than or equal to PID min value'